src/Insolation.jl

```diff
@@ -138,6 +139,86 @@ function orbital_params(param_set::AIP)
     return ϖ, γ, e
 end
 
+"""
+    TSIDataSpline
+
+A container struct that holds the monthly mean total solar irradiance.
+
+The spline is a function of date between `1850-01-01` and
+`2299-12-15`.
+
+# GPU Support
+This struct is GPU-compatible via Adapt.jl. To transfer to GPU memory:
+```julia
+using CUDA, Adapt
+cpu_tsi = TSIDataSpline(Float32)  # Create on CPU
+gpu_tsi = adapt(CuArray, TSIDataSpline)  # Transfer to GPU
+```
+"""
+struct TSIDataSpline{SPLINE,REF_DATE<:Dates.AbstractDateTime}
+    tsi_spline::SPLINE
+    ref_date::REF_DATE
+end
+
+"""
+    _get_tsi_data()
+
+Return the monthly dates and TSI data from the `cmip_monthly_tsi` artifact.
+"""
+function _get_tsi_data()
+    datapath = joinpath(artifact"cmip_monthly_tsi", "cmip_monthly_tsi.csv")
+    monthly_tsi_data, _ = readdlm(datapath, ',', String, header = true)
+
+    # For implementing evaluate, it is simpler to work with the first day of the
+    # month instead of the middle of the month
+    monthly_dates = Dates.firstdayofmonth.(Dates.DateTime.(monthly_tsi_data[:, 1]))
+    tsi_data = parse.(Float32, monthly_tsi_data[:, 2])
+    return monthly_dates, tsi_data
+end
+
+"""
+    TSIDataSpline(::Type{FT}) where {FT <: AbstractFloat}
+
+Constructs a `TSIDataSpline` that linearly interpolates monthly mean total solar
```

Comment:
```suggestion
Constructs a `TSIDataSpline` that interpolates monthly mean total solar
```
(Looks like it's cubic interpolation now.)